lib/dnsimple/client/certificates.rb

```diff
@@ -108,6 +108,167 @@ def certificate_private_key(account_id, domain_id, certificate_id, options = {})
         Dnsimple::Response.new(response, Struct::CertificateBundle.new(response["data"]))
       end
 
+      # Purchase a Let's Encrypt certificate.
+      #
+      # This method creates a new certificate order. The certificate ID should be used to
+      # request the issuance of the certificate using {#letsencrypt_issue}.
+      #
+      # @see https://developer.dnsimple.com/v2/domains/certificates/#letsencrypt-purchase
+      #
+      # @param  account_id [Integer] the account ID
+      # @param  domain_id [#to_s] the domain ID or domain name
+      # @param  attributes [Hash]
+      # @option attributes [Integer] :contact_id the contact ID (mandatory)
+      # @option attributes [String] :name the certificate name (optional)
```

Comment:
AFAIK the name should be mandatory (as we do for the record). If for the root domain, it should be an empty string "".

Reply:
If `:name` is omitted we consider root certificate.

Reply:
Is it something we do in the app? I think we should validate/enforce it as for the records to avoid incorrectly created certificates.